### ISSUE TITLE
CPU spike with VST3 plugins.

### ISSUE THREAD

**Issue by Stazed** [lsp-plugins/lsp-plugins#409]
After getting VST3 to work with LSP in non-mixer-xt, I noticed a significant sustained CPU spiking
occurring when loading any VST3 plugin that was not present in the other types.

Building from git Master 1.2.16.

I managed to locate the origin of the spike in:
```
/lsp-plugins/modules/lsp-plugin-fw/include/lsp-plug.in/plug-fw/wrap/vst3/impl/factory.h

Line 733: status_t PluginFactory::run() {}
```
 
Line 776 of the function logic seems backward? Looks like it should be 'lsp_max' instead 
of 'lsp_min' for the minimum 'delay' of 40U.

```
-- const system::time_millis_t delay = lsp_min(system::get_time_millis() - time, 40u);
```
```
++ const system::time_millis_t delay = lsp_max(system::get_time_millis() - time, 40u);
```

Changing the code as above fixed the CPU spiking and as far as I can see, with no side effects.
Thanks!

**Comment by Stazed** [comment]
Actually the above does not seem right either, though it might work if (system::get_time_millis() - time)
is always less than 40u.

Maybe:

```
const system::time_millis_t delay = (system::get_time_millis() - time) < 40u ?  40u : 0;
```
Tried this and it also seems to eliminate the sustained spike.

**Comment by sadko4u** [comment]
Oh yes, it seems that time interval is improperly calculated. If we consume around 1 ms for the loop, then we need to sleep the rest 39 milliseconds, not 1 as it is calculated at this moment.

**Comment by sadko4u** [comment]
Fixed in this commit:
https://github.com/lsp-plugins/lsp-plugin-fw/commit/e615e6db15e666b4424304f67359228eec1ae879

**Comment by sadko4u** [comment]
Fixed in 1.2.17.
